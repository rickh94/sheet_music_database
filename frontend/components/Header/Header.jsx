import React, { Component } from 'react'
import Navbar from 'react-bulma-components/lib/components/navbar'
import Heading from 'react-bulma-components/lib/components/heading'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import './Header.scss'

export class Header extends Component {
  state = {
    open: false
  }

  static propTypes = {
    account: PropTypes.object.isRequired
  }

  render() {
    const { open } = this.state
    const { token } = this.props.account
    const menuActive = open ? 'is-active' : ''
    return (
      <Navbar color="primary" textColor="light">
        <Navbar.Brand
          textColor="white"
          renderAs="div"
          paddingless={true}
          marginless={true}
        >
          <Link to="/" className="header--link">
            <Navbar.Item
              renderAs="span"
              paddingless={true}
              marginless={true}
              style={{ height: '70%' }}
            >
              <Heading
                size={5}
                renderAs="div"
                style={{ verticalAlign: 'middle', color: '#f0f0f0' }}
                marginless={true}
                paddingless={true}
              >
                Sheet Music Database
              </Heading>
            </Navbar.Item>
          </Link>
          <Navbar.Burger
            className={menuActive}
            paddingless={true}
            marginless={true}
            onClick={() => this.setState({ open: !open })}
          />
        </Navbar.Brand>
        <Navbar.Menu className={menuActive} color="primary">
          <Navbar.Container color="primary" position="end">
            <HeaderLink to="/">Home</HeaderLink>
            <HeaderLink to="/tags">Tags</HeaderLink>
            {token ? (
              <React.Fragment>
                <HeaderLink to="/profile">Profile</HeaderLink>
                <HeaderLink to="/logout">Logout</HeaderLink>
              </React.Fragment>
            ) : (
              <React.Fragment>
                <HeaderLink to="/register">Register</HeaderLink>
                <HeaderLink to="/login">Login</HeaderLink>
              </React.Fragment>
            )}
          </Navbar.Container>
        </Navbar.Menu>
      </Navbar>
    )
  }
}

const mapStateToProps = state => {
  return {
    account: state.account
  }
}

const mapDispatchToProps = dispatch => {
  return {}
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header)

export function HeaderLink(props) {
  return (
    <Link to={props.to} className="header--link">
      <Navbar.Item renderAs="div" className="nav-item-resp">
        {props.children}
      </Navbar.Item>
    </Link>
  )
}

HeaderLink.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired
}
