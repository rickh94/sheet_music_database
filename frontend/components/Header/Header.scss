$desktop: 1024px;

a {
  text-decoration: none;
}

.nav-item-resp {
  @media only screen and (min-width: $desktop) {
      color: #f0f0f0 !important;
      vertical-align: middle;
      height: 70%;
  }
}

