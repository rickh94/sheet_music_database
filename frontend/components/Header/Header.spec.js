import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'
import '../../setupTests'
import Navbar from 'react-bulma-components/lib/components/navbar'

describe('Header', () => {
  const wrapper = shallow(<Header />)
  it('renders a Navbar', () => {
    expect(wrapper.find('Navbar').length).toEqual(1)
  })

  it('renders a Brand', () => {
    expect(wrapper.containsMatchingElement(<Navbar.Brand>Sheet Music Database</Navbar.Brand>)).toBeTruthy()
  })

  it('renders a Burger', () => {
    expect(wrapper.containsMatchingElement(<Navbar.Burger active={wrapper.instance().state.open} />)).toBeTruthy()
  })

  it('renders a menu', () => {
    expect(wrapper.find('#nav-menu').length).toBeTruthy()
  })
})
