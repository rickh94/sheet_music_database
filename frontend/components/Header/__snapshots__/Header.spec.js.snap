// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    account={
      Object {
        "token": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "backgroundColor": undefined,
      "children": Array [
        <NavbarBrand
          className=""
          marginless={true}
          paddingless={true}
          renderAs="div"
          style={Object {}}
          textColor="white"
        >
          <Link
            className="header--link"
            replace={false}
            to="/"
          >
            <NavbarItem
              active={false}
              className=""
              dropdown={false}
              dropdownUp={false}
              hoverable={false}
              marginless={true}
              paddingless={true}
              renderAs="span"
              style={
                Object {
                  "height": "70%",
                }
              }
            >
              <Heading
                className=""
                heading={false}
                marginless={true}
                paddingless={true}
                renderAs="div"
                size={5}
                spaced={false}
                style={
                  Object {
                    "color": "#f0f0f0",
                    "verticalAlign": "middle",
                  }
                }
                subtitle={false}
                weight={null}
              >
                Sheet Music Database
              </Heading>
            </NavbarItem>
          </Link>
          <NavbarBurger
            className=""
            marginless={true}
            onClick={[Function]}
            paddingless={true}
            style={Object {}}
          />
        </NavbarBrand>,
        <NavbarMenu
          className=""
          color="primary"
          style={Object {}}
        >
          <NavbarContainer
            className=""
            color="primary"
            position="end"
            renderAs="div"
            style={Object {}}
          >
            <HeaderLink
              to="/"
            >
              Home
            </HeaderLink>
            <HeaderLink
              to="/tags"
            >
              Tags
            </HeaderLink>
            <React.Fragment>
              <HeaderLink
                to="/register"
              >
                Register
              </HeaderLink>
              <HeaderLink
                to="/login"
              >
                Login
              </HeaderLink>
            </React.Fragment>
          </NavbarContainer>
        </NavbarMenu>,
      ],
      "className": "",
      "clearfix": undefined,
      "clipped": undefined,
      "color": "primary",
      "fixed": null,
      "hidden": undefined,
      "invisible": undefined,
      "italic": undefined,
      "marginless": undefined,
      "overlay": undefined,
      "paddingless": undefined,
      "pull": undefined,
      "radiusless": undefined,
      "renderAs": "nav",
      "responsive": undefined,
      "shadowless": undefined,
      "style": Object {},
      "textAlignment": undefined,
      "textColor": "light",
      "textSize": undefined,
      "textTransform": undefined,
      "textWeight": undefined,
      "transparent": false,
      "unselectable": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "children": Array [
            <Link
              className="header--link"
              replace={false}
              to="/"
            >
              <NavbarItem
                active={false}
                className=""
                dropdown={false}
                dropdownUp={false}
                hoverable={false}
                marginless={true}
                paddingless={true}
                renderAs="span"
                style={
                  Object {
                    "height": "70%",
                  }
                }
              >
                <Heading
                  className=""
                  heading={false}
                  marginless={true}
                  paddingless={true}
                  renderAs="div"
                  size={5}
                  spaced={false}
                  style={
                    Object {
                      "color": "#f0f0f0",
                      "verticalAlign": "middle",
                    }
                  }
                  subtitle={false}
                  weight={null}
                >
                  Sheet Music Database
                </Heading>
              </NavbarItem>
            </Link>,
            <NavbarBurger
              className=""
              marginless={true}
              onClick={[Function]}
              paddingless={true}
              style={Object {}}
            />,
          ],
          "className": "",
          "clearfix": undefined,
          "clipped": undefined,
          "hidden": undefined,
          "invisible": undefined,
          "italic": undefined,
          "marginless": true,
          "overlay": undefined,
          "paddingless": true,
          "pull": undefined,
          "radiusless": undefined,
          "renderAs": "div",
          "responsive": undefined,
          "shadowless": undefined,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": "white",
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <NavbarItem
                active={false}
                className=""
                dropdown={false}
                dropdownUp={false}
                hoverable={false}
                marginless={true}
                paddingless={true}
                renderAs="span"
                style={
                  Object {
                    "height": "70%",
                  }
                }
              >
                <Heading
                  className=""
                  heading={false}
                  marginless={true}
                  paddingless={true}
                  renderAs="div"
                  size={5}
                  spaced={false}
                  style={
                    Object {
                      "color": "#f0f0f0",
                      "verticalAlign": "middle",
                    }
                  }
                  subtitle={false}
                  weight={null}
                >
                  Sheet Music Database
                </Heading>
              </NavbarItem>,
              "className": "header--link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "backgroundColor": undefined,
                "children": <Heading
                  className=""
                  heading={false}
                  marginless={true}
                  paddingless={true}
                  renderAs="div"
                  size={5}
                  spaced={false}
                  style={
                    Object {
                      "color": "#f0f0f0",
                      "verticalAlign": "middle",
                    }
                  }
                  subtitle={false}
                  weight={null}
                >
                  Sheet Music Database
                </Heading>,
                "className": "",
                "clearfix": undefined,
                "clipped": undefined,
                "dropdown": false,
                "dropdownUp": false,
                "hidden": undefined,
                "hoverable": false,
                "invisible": undefined,
                "italic": undefined,
                "marginless": true,
                "overlay": undefined,
                "paddingless": true,
                "pull": undefined,
                "radiusless": undefined,
                "renderAs": "span",
                "responsive": undefined,
                "shadowless": undefined,
                "style": Object {
                  "height": "70%",
                },
                "textAlignment": undefined,
                "textColor": undefined,
                "textSize": undefined,
                "textTransform": undefined,
                "textWeight": undefined,
                "unselectable": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "children": "Sheet Music Database",
                  "className": "",
                  "clearfix": undefined,
                  "clipped": undefined,
                  "heading": false,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": true,
                  "overlay": undefined,
                  "paddingless": true,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "div",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "size": 5,
                  "spaced": false,
                  "style": Object {
                    "color": "#f0f0f0",
                    "verticalAlign": "middle",
                  },
                  "subtitle": false,
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                  "weight": null,
                },
                "ref": null,
                "rendered": "Sheet Music Database",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "className": "",
              "clearfix": undefined,
              "clipped": undefined,
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": true,
              "onClick": [Function],
              "overlay": undefined,
              "paddingless": true,
              "pull": undefined,
              "radiusless": undefined,
              "responsive": undefined,
              "shadowless": undefined,
              "style": Object {},
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "children": <NavbarContainer
            className=""
            color="primary"
            position="end"
            renderAs="div"
            style={Object {}}
          >
            <HeaderLink
              to="/"
            >
              Home
            </HeaderLink>
            <HeaderLink
              to="/tags"
            >
              Tags
            </HeaderLink>
            <React.Fragment>
              <HeaderLink
                to="/register"
              >
                Register
              </HeaderLink>
              <HeaderLink
                to="/login"
              >
                Login
              </HeaderLink>
            </React.Fragment>
          </NavbarContainer>,
          "className": "",
          "clearfix": undefined,
          "clipped": undefined,
          "color": "primary",
          "hidden": undefined,
          "invisible": undefined,
          "italic": undefined,
          "marginless": undefined,
          "overlay": undefined,
          "paddingless": undefined,
          "pull": undefined,
          "radiusless": undefined,
          "responsive": undefined,
          "shadowless": undefined,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": undefined,
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": Array [
              <HeaderLink
                to="/"
              >
                Home
              </HeaderLink>,
              <HeaderLink
                to="/tags"
              >
                Tags
              </HeaderLink>,
              <React.Fragment>
                <HeaderLink
                  to="/register"
                >
                  Register
                </HeaderLink>
                <HeaderLink
                  to="/login"
                >
                  Login
                </HeaderLink>
              </React.Fragment>,
            ],
            "className": "",
            "clearfix": undefined,
            "clipped": undefined,
            "color": "primary",
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "position": "end",
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Home",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Tags",
                "to": "/tags",
              },
              "ref": null,
              "rendered": "Tags",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <HeaderLink
                    to="/register"
                  >
                    Register
                  </HeaderLink>,
                  <HeaderLink
                    to="/login"
                  >
                    Login
                  </HeaderLink>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Register",
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": "Register",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Login",
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
              ],
              "type": Symbol(react.fragment),
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "backgroundColor": undefined,
        "children": Array [
          <NavbarBrand
            className=""
            marginless={true}
            paddingless={true}
            renderAs="div"
            style={Object {}}
            textColor="white"
          >
            <Link
              className="header--link"
              replace={false}
              to="/"
            >
              <NavbarItem
                active={false}
                className=""
                dropdown={false}
                dropdownUp={false}
                hoverable={false}
                marginless={true}
                paddingless={true}
                renderAs="span"
                style={
                  Object {
                    "height": "70%",
                  }
                }
              >
                <Heading
                  className=""
                  heading={false}
                  marginless={true}
                  paddingless={true}
                  renderAs="div"
                  size={5}
                  spaced={false}
                  style={
                    Object {
                      "color": "#f0f0f0",
                      "verticalAlign": "middle",
                    }
                  }
                  subtitle={false}
                  weight={null}
                >
                  Sheet Music Database
                </Heading>
              </NavbarItem>
            </Link>
            <NavbarBurger
              className=""
              marginless={true}
              onClick={[Function]}
              paddingless={true}
              style={Object {}}
            />
          </NavbarBrand>,
          <NavbarMenu
            className=""
            color="primary"
            style={Object {}}
          >
            <NavbarContainer
              className=""
              color="primary"
              position="end"
              renderAs="div"
              style={Object {}}
            >
              <HeaderLink
                to="/"
              >
                Home
              </HeaderLink>
              <HeaderLink
                to="/tags"
              >
                Tags
              </HeaderLink>
              <React.Fragment>
                <HeaderLink
                  to="/register"
                >
                  Register
                </HeaderLink>
                <HeaderLink
                  to="/login"
                >
                  Login
                </HeaderLink>
              </React.Fragment>
            </NavbarContainer>
          </NavbarMenu>,
        ],
        "className": "",
        "clearfix": undefined,
        "clipped": undefined,
        "color": "primary",
        "fixed": null,
        "hidden": undefined,
        "invisible": undefined,
        "italic": undefined,
        "marginless": undefined,
        "overlay": undefined,
        "paddingless": undefined,
        "pull": undefined,
        "radiusless": undefined,
        "renderAs": "nav",
        "responsive": undefined,
        "shadowless": undefined,
        "style": Object {},
        "textAlignment": undefined,
        "textColor": "light",
        "textSize": undefined,
        "textTransform": undefined,
        "textWeight": undefined,
        "transparent": false,
        "unselectable": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": Array [
              <Link
                className="header--link"
                replace={false}
                to="/"
              >
                <NavbarItem
                  active={false}
                  className=""
                  dropdown={false}
                  dropdownUp={false}
                  hoverable={false}
                  marginless={true}
                  paddingless={true}
                  renderAs="span"
                  style={
                    Object {
                      "height": "70%",
                    }
                  }
                >
                  <Heading
                    className=""
                    heading={false}
                    marginless={true}
                    paddingless={true}
                    renderAs="div"
                    size={5}
                    spaced={false}
                    style={
                      Object {
                        "color": "#f0f0f0",
                        "verticalAlign": "middle",
                      }
                    }
                    subtitle={false}
                    weight={null}
                  >
                    Sheet Music Database
                  </Heading>
                </NavbarItem>
              </Link>,
              <NavbarBurger
                className=""
                marginless={true}
                onClick={[Function]}
                paddingless={true}
                style={Object {}}
              />,
            ],
            "className": "",
            "clearfix": undefined,
            "clipped": undefined,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": true,
            "overlay": undefined,
            "paddingless": true,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": "white",
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <NavbarItem
                  active={false}
                  className=""
                  dropdown={false}
                  dropdownUp={false}
                  hoverable={false}
                  marginless={true}
                  paddingless={true}
                  renderAs="span"
                  style={
                    Object {
                      "height": "70%",
                    }
                  }
                >
                  <Heading
                    className=""
                    heading={false}
                    marginless={true}
                    paddingless={true}
                    renderAs="div"
                    size={5}
                    spaced={false}
                    style={
                      Object {
                        "color": "#f0f0f0",
                        "verticalAlign": "middle",
                      }
                    }
                    subtitle={false}
                    weight={null}
                  >
                    Sheet Music Database
                  </Heading>
                </NavbarItem>,
                "className": "header--link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "active": false,
                  "backgroundColor": undefined,
                  "children": <Heading
                    className=""
                    heading={false}
                    marginless={true}
                    paddingless={true}
                    renderAs="div"
                    size={5}
                    spaced={false}
                    style={
                      Object {
                        "color": "#f0f0f0",
                        "verticalAlign": "middle",
                      }
                    }
                    subtitle={false}
                    weight={null}
                  >
                    Sheet Music Database
                  </Heading>,
                  "className": "",
                  "clearfix": undefined,
                  "clipped": undefined,
                  "dropdown": false,
                  "dropdownUp": false,
                  "hidden": undefined,
                  "hoverable": false,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": true,
                  "overlay": undefined,
                  "paddingless": true,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "span",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "style": Object {
                    "height": "70%",
                  },
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "backgroundColor": undefined,
                    "children": "Sheet Music Database",
                    "className": "",
                    "clearfix": undefined,
                    "clipped": undefined,
                    "heading": false,
                    "hidden": undefined,
                    "invisible": undefined,
                    "italic": undefined,
                    "marginless": true,
                    "overlay": undefined,
                    "paddingless": true,
                    "pull": undefined,
                    "radiusless": undefined,
                    "renderAs": "div",
                    "responsive": undefined,
                    "shadowless": undefined,
                    "size": 5,
                    "spaced": false,
                    "style": Object {
                      "color": "#f0f0f0",
                      "verticalAlign": "middle",
                    },
                    "subtitle": false,
                    "textAlignment": undefined,
                    "textColor": undefined,
                    "textSize": undefined,
                    "textTransform": undefined,
                    "textWeight": undefined,
                    "unselectable": undefined,
                    "weight": null,
                  },
                  "ref": null,
                  "rendered": "Sheet Music Database",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": undefined,
                "className": "",
                "clearfix": undefined,
                "clipped": undefined,
                "hidden": undefined,
                "invisible": undefined,
                "italic": undefined,
                "marginless": true,
                "onClick": [Function],
                "overlay": undefined,
                "paddingless": true,
                "pull": undefined,
                "radiusless": undefined,
                "responsive": undefined,
                "shadowless": undefined,
                "style": Object {},
                "textAlignment": undefined,
                "textColor": undefined,
                "textSize": undefined,
                "textTransform": undefined,
                "textWeight": undefined,
                "unselectable": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": <NavbarContainer
              className=""
              color="primary"
              position="end"
              renderAs="div"
              style={Object {}}
            >
              <HeaderLink
                to="/"
              >
                Home
              </HeaderLink>
              <HeaderLink
                to="/tags"
              >
                Tags
              </HeaderLink>
              <React.Fragment>
                <HeaderLink
                  to="/register"
                >
                  Register
                </HeaderLink>
                <HeaderLink
                  to="/login"
                >
                  Login
                </HeaderLink>
              </React.Fragment>
            </NavbarContainer>,
            "className": "",
            "clearfix": undefined,
            "clipped": undefined,
            "color": "primary",
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "children": Array [
                <HeaderLink
                  to="/"
                >
                  Home
                </HeaderLink>,
                <HeaderLink
                  to="/tags"
                >
                  Tags
                </HeaderLink>,
                <React.Fragment>
                  <HeaderLink
                    to="/register"
                  >
                    Register
                  </HeaderLink>
                  <HeaderLink
                    to="/login"
                  >
                    Login
                  </HeaderLink>
                </React.Fragment>,
              ],
              "className": "",
              "clearfix": undefined,
              "clipped": undefined,
              "color": "primary",
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": undefined,
              "overlay": undefined,
              "paddingless": undefined,
              "position": "end",
              "pull": undefined,
              "radiusless": undefined,
              "renderAs": "div",
              "responsive": undefined,
              "shadowless": undefined,
              "style": Object {},
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Home",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Tags",
                  "to": "/tags",
                },
                "ref": null,
                "rendered": "Tags",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <HeaderLink
                      to="/register"
                    >
                      Register
                    </HeaderLink>,
                    <HeaderLink
                      to="/login"
                    >
                      Login
                    </HeaderLink>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Register",
                      "to": "/register",
                    },
                    "ref": null,
                    "rendered": "Register",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Login",
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                ],
                "type": Symbol(react.fragment),
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
